### PR TITLE
Fix simultaneous network requests for same url not working

### PR DESCRIPTION
**Pull request by esad**

### CODE REVIEW

File: Carlos/NetworkFetcher.swift [hunk 1]
```diff
@@ -65,15 +65,42 @@ public class NetworkFetcher: CacheLevel {
...
+        fail(errorWithCode(code.rawValue),self)
       }
     }
   }
-  
-  private var pendingRequests: [String: Request] = [:]
-  
+
+  private lazy var lockQueue: dispatch_queue_t = {
+    return dispatch_queue_create(CarlosGlobals.QueueNamePrefix + "NetworkFetcher", nil)
```
Comment (vittoriom) [inline Carlos/NetworkFetcher.swift]: can we make it explicit that this is a serial queue?

File: Carlos/NetworkFetcher.swift [hunk 2]
```diff
@@ -65,15 +65,42 @@ public class NetworkFetcher: CacheLevel {
...
+    dispatch_async(lockQueue) {
+      self.pendingRequests.append(request)
+    }
+  }
+
+  private func removePendingRequests(request: Request) {
+    dispatch_async(lockQueue) {
+      var idx: Int?
+
+      for (index, obj) in enumerate(self.pendingRequests) {
```
Comment (vittoriom) [inline Carlos/NetworkFetcher.swift]: isn't there any way to find the index via the "find" function we have in Swift? or something less low-level?

File: Tests/NetworkFetcherTests.swift [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+class NetworkFetcherTests: QuickSpec {
+  override func spec() {
+    describe("NetworkFetcher") {
+      var sut: NetworkFetcher!
+
+      beforeEach {
+        sut = NetworkFetcher()
+      }
+
+      context("simulatenous requests") {
```
Comment (vittoriom) [inline Tests/NetworkFetcherTests.swift]: some typo here "simultaneous"

File: Tests/NetworkFetcherTests.swift [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        var finished = 0
+        let simulatenousRequests = 3
```
Comment (vittoriom) [inline Tests/NetworkFetcherTests.swift]: same here

File: Carlos/NetworkFetcher.swift (hunk 2 above)
Comment (esad) [inline Carlos/NetworkFetcher.swift]: I couldn't find anything that works with object identity (===). There's indexOf accepting a closure in Swift 2..

Comment (vittoriom) [inline Carlos/NetworkFetcher.swift]: I see. Then at least let's do something like:

`let idx = filter(enumerate(self.pendingRequests), { $1 === request }).first?.index`

**Pull request merged**